waspc/data/Generator/templates/sdk/wasp/server/auth/hooks.ts

```diff
@@ -0,0 +1,46 @@
+import type { Request as ExpressRequest } from 'express'
+import type { ProviderId, createUser } from '../../auth/utils.js'
+import { prisma } from '../index.js'
+
+type CommonInput = {
+  hookName: string
+  prisma: typeof prisma
+  req: ExpressRequest
+}
+
+/* On Before Signup Hook */
+export type OnBeforeSignupHookFn = (
```

Comment:
I haven't added the ability for developers to modify the data that's used to create the `user`. I'm not sure if it's useful since we also have `userSignupFields` 🤷

Reply:
Let's add it when it proves necessary.